src/workerd/io/actor-cache.h

```diff
@@ -427,11 +434,23 @@ class ActorCache final: public ActorCacheInterface {
       }
     }
 
-    // This enum indicates how synchronized this entry is with storage.
-    EntrySyncStatus syncStatus = EntrySyncStatus::NOT_IN_CACHE;
+    void setFlushing() {
+      syncStatus = EntrySyncStatus::FLUSHING;
+    }
+
+    void setNotInCache() {
+      syncStatus = EntrySyncStatus::NOT_IN_CACHE;
+    }
+
+    // Avoid using this directly. If you want to set the status to CLEAN or DIRTY, consider using
```

Comment:
If we have this function, but we're not supposed to use it, that's an indication of something buggy in our abstraction layers.

A better approach might be to remove `setSyncStatus` and have setClean and setDirty instead.

Reply:
What we really want is to prevent directly modifying the `syncStatus` unless it's part of some action (like adding to a list), but the methods that add to the respective lists don't have direct access to the `syncStatus`. The same issue would exist with `setClean` and `setDirty` -- though I suppose it would make it more obvious that you'd have to use `setNotInCache()` explicitly too...